### ISSUE TITLE
Intellisense errors

### ISSUE THREAD

**Issue by ImanHosseini** [redhat-developer/vscode-java#2839]
When I do "./gradlew build" -> it works. no problem. 
BUT intellisense shows errors like:
```
The import com.sun.javadoc cannot be resolved
The import com.sun.tools cannot be resolved
```
The project I work with is not meant for java newer than 1.8. In my workspace's .vscode/settings.json I have:
```
{
    "java.configuration.updateBuildConfiguration": "interactive",
    "java.jdt.ls.java.home" :"/usr/lib/jvm/java-1.8.0-openjdk-amd64",
    "java.import.gradle.java.home" : "/usr/lib/jvm/java-1.8.0-openjdk-amd64"
}
```
##### Environment
- Operating System: WSL2
- JDK version: 
- Visual Studio Code version: 1.74.1
- Java extension version: 1.13.0

**Comment by jdneo** [comment]
Can you remove `"java.jdt.ls.java.home" :"/usr/lib/jvm/java-1.8.0-openjdk-amd64"` from your settings and try again?

**Comment by ImanHosseini** [comment]
Yes, it doesn't fix it. Actually initially I didn't have it in settings, I added that thinking it will fix the issue (because when I compile with newer java, I get an error regarding javadoc, but with 1.8 it works -> so I thought the squiggly line would go away if I somehow let the extension know that I want Java 1.8).

**Comment by ImanHosseini** [comment]
This might be related and a hint as to what is going wrong:
![rd](https://user-images.githubusercontent.com/12172889/208107933-e071d041-8b0e-4dfb-b72c-34f112de5013.png)
And when I click on it I get: The editor could not be opened due to an unexpected error: File is a directory
Now what is this 'tools.jar'? The only place I see it is in a build.gradle file - in this line:
```
compile files(System.getenv('JAVA_HOME') + '/lib/tools.jar')
```
Which is again very weird! Here is what I get from terminal:
```
$echo $JAVA_HOME

```
Nothing -> so why is that env var messed up? And what is that "null" doing there?

**Comment by snjeza** [comment]
@ImanHosseini You can try the following:

- add
```
dependencies {
...
	implementation files("${System.properties['java.home']}/../lib/tools.jar")
...
}
```
to your build.gradle 
- run VS Code
```
$ export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-amd64
$ code .
```

**Comment by ImanHosseini** [comment]
I don't know what is causing this but this is absurd! Even if I hardcode that to:
```
files("/usr/lib/jvm/java-1.8.0-openjdk-amd64/lib/tools.jar")
```
- and this is THE ONLY place in the whole project that tools.jar shows up - I still get that error! (tried restarting vscode/ extenstion etc.)
The errors:
```
The container 'Project and External Dependencies' references non existing library '/home/iman/jco/battlecode22/engine/null/lib/tools.jar' [ln 1, Col1]
```
It's also weird what is Ln1 Col1 mean? That line is not 1st line, whatever is causing this, the error is also bad because instead of a file it points to a directory (thus another error if you click on it) and it doesn't help find what is _really_ wrong here.

**Comment by snjeza** [comment]
@ImanHosseini Could you try to install `java-1.8.0-openjdk-devel` ?

**Comment by ImanHosseini** [comment]
I'm on Ubuntu and I have openjdk-8-jdk package. 'java-1.8.0-openjdk-devel' is not on Ubuntu repos am I right?

**Comment by ImanHosseini** [comment]
I also get the "Configure Runtime for Language Server" show up
![rrr](https://user-images.githubusercontent.com/12172889/208204513-ead5b514-4d84-4246-9adf-47ba22021373.png)
 whenever I open vscode and the 'Reload' button does not work. It just does not do anything.

**Comment by snjeza** [comment]
@ImanHosseini Could you try 
```
$ ls /usr/lib/jvm/java-1.8.0-openjdk-amd64/lib/tools.jar
```

**Comment by ImanHosseini** [comment]
yes. I did check it, tools.jar is indeed in that directory. It was, I think, only an issue in pre-release though. Because now I went back to v1.13.0 and reloaded, this issue is now fixed. The other issue persists though: every time I open vscode, I get the "Configure Runtime for Language Server" pop up, and the reload button on it does not work after I set my path. And the intellisense starts being fine nonetheless - something triggers this to pop up, but at the same time I see "extension activating" and then after a few minutes extension activates and is fine. So I think the issue here is, that pop-up is being triggered prematurely - before extension is finished activating. But anyway, this issue is just a nuisance but at least it works.

**Comment by snjeza** [comment]
@ImanHosseini VS Code requires JDK 17 - https://github.com/redhat-developer/vscode-java/wiki/JDK-Requirements
Could you try to remove
```
 "java.jdt.ls.java.home" :"/usr/lib/jvm/java-1.8.0-openjdk-amd64",
```

**Comment by afarah1** [comment]
For me the error ("The container 'Project and External Dependencies' references non existing library" pointing to a lib under ".gradle/caches") on vscode occurred after upgrading from java 11 to java 17. I had java installed with homebrew on a mac on a custom folder under my home. 

I added the new java 17 install location to my PATH, and also to `"java.import.gradle.java.home"` on vscode. I also removed `~/.gradle/caches` and restarted vscode. I'm not sure if all these steps are necessary but it solved for me. I had the issue again when I inadvertently removed the PATH and vscode configs, re-doing all the steps solved it again.
